test/proxy-call-test.js

```diff
@@ -1176,29 +1176,44 @@ describe("sinonSpy.call", function () {
 
         // https://github.com/sinonjs/sinon/issues/1066
         /* eslint-disable consistent-return */
-        it("does not throw when the call stack is empty", function (done) {
+        it("does not throw when the call stack is empty", async function () {
             if (typeof Promise === "undefined") {
                 this.skip();
             }
 
             var stub1 = sinonStub().resolves(1);
             var stub2 = sinonStub().returns(1);
 
-            function run() {
-                return stub1().then(stub2);
-            }
+            await stub2(await stub1());
```

Comment:
This could be
```
const val = await stub1();
await stub2(val);
```
if that seems better, I couldn't decide which was less confusing 😄

Reply:
TBH I find Promise chaining much more understandable, as in `await stub1().then(stub2)`, but I do not really care all that much either :p Whatever rocks your boat :)